Fix long line splitting issue in python_print

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36188 Fix long line splitting issue in python_print**

Differential Revision: [D20907254](https://our.internmc.facebook.com/intern/diff/D20907254)

zdevito committed:
Fix long line splitting issue in python_print

* Need to remove n^2 behavior for scanning whether to split or not
  otherwise long inline chains will take a long time re-scanning.

[ghstack-poisoned]

zdevito committed:
Update on "Fix long line splitting issue in python_print"


* Need to remove n^2 behavior for scanning whether to split or not
  otherwise long inline chains will take a long time re-scanning.

Differential Revision: [D20907254](https://our.internmc.facebook.com/intern/diff/D20907254)

[ghstack-poisoned]

zdevito committed:
Update on "Fix long line splitting issue in python_print"


* Need to remove n^2 behavior for scanning whether to split or not
  otherwise long inline chains will take a long time re-scanning.

Differential Revision: [D20907254](https://our.internmc.facebook.com/intern/diff/D20907254)

[ghstack-poisoned]

zdevito committed:
Update on "Fix long line splitting issue in python_print"


* Need to remove n^2 behavior for scanning whether to split or not
  otherwise long inline chains will take a long time re-scanning.

Differential Revision: [D20907254](https://our.internmc.facebook.com/intern/diff/D20907254)

[ghstack-poisoned]

zdevito committed:
Update on "Fix long line splitting issue in python_print"


* Need to remove n^2 behavior for scanning whether to split or not
  otherwise long inline chains will take a long time re-scanning.

Differential Revision: [D20907254](https://our.internmc.facebook.com/intern/diff/D20907254)

[ghstack-poisoned]

zdevito committed:
Update on "Fix long line splitting issue in python_print"


* Need to remove n^2 behavior for scanning whether to split or not
  otherwise long inline chains will take a long time re-scanning.

Differential Revision: [D20907254](https://our.internmc.facebook.com/intern/diff/D20907254)

[ghstack-poisoned]

suo reviewed (approved):
lol, maybe inlining isn't worth…

zdevito committed:
Update on "Fix long line splitting issue in python_print"


Differential Revision: [D20907254](https://our.internmc.facebook.com/intern/diff/D20907254)

[ghstack-poisoned]

mruberry commented:
Unlanding. This broke the clang-format build. You can run tools/clang_format.py before committing.
